docs/migrating-to-v6.md

```diff
@@ -231,6 +242,81 @@ The OAuth methods still behave the same way, but we've updated their signatures
 
 1. There is a new `shopify.session` object which contains session-specific functions. See the [Utility functions](#utility-functions) section for the specific changes.
 
+## Billing
+
+The billing functionality hasn't changed, but the main difference is that the library now provides separate methods for checking and requesting payment, which gives apps more freedom to implement their billing logic.
+
+To configure billing, you can now pass in more than one billing plan, and they're indexed by plan name:
+
+<div>Before
+
+```ts
+Shopify.Context.initialize({
+  // ...
+  billing: {
+    chargeName: 'My plan',
+    amount: 5.0,
+    currencyCode: 'USD',
+    interval: BillingInterval.Every30Days,
+  },
+});
+```
+
+</div><div>After
+
+```ts
+const shopify = shopifyApi({
+  // ...
+  billing: {
+    'My plan': {
+      amount: 5.0,
+      currencyCode: 'USD',
+      interval: BillingInterval.Every30Days,
+    },
+  },
+});
+```
+
+</div>
+
+We broke the `Shopify.Billing.check` method up into `shopify.billing.check` and `shopify.billing.request`, as mentioned above:
+
+<div>Before
+
+```ts
+const {hasPayment, confirmBillingUrl} = await Shopify.Billing.check({
+  session,
+  isTest: true,
+});
+
+if (!hasPayment) {
+  return redirect(confirmBillingUrl);
+}
+```
+
+</div><div>After
+
+```ts
+const hasPayment = await shopify.billing.check({
+  session,
+  plans: 'My plan',
+  isTest: true,
+});
+
+if (!hasPayment) {
+  const confirmationUrl = await shopify.billing.request({
+    session,
+    plan: 'My plan',
+    isTest: true,
+  });
+  return redirect(confirmationUrl);
+}
+```
+
+Note that when calling `check`, apps can pass in one or more plans, and it will return true if any of them match. The `request` method creates a charge using the given plan's configs.
```

Comment:
```suggestion
Note that when calling `check`, apps can pass in one or more plans, and it will return true if any of them match. The `request` method creates a charge using the configuration of the given plan name.
```